docs/source/examplecleanup.rst

```diff
@@ -6,55 +6,106 @@ End-to-End Example
 
 An End-to-End example walkthrough of a tartufo scan and the process of purging the dirty evil passwords that somehow ended up in your code commits. We will use an additional tool: ``BFG`` (https://rtyley.github.io/bfg-repo-cleaner/, more on this later!). 
 
-#. Use ``tartufo`` to scan your repository and find any secrets in its history!
 
-   Use what you've learned to scan your repo!
+#. OPTIONAL Development only: Setup poetry if you want to use the most recent non-released build from github (may not be stable)
+
+   This project uses `Poetry`_ to manage its dependencies and do a lot of the heavy lifting. So you'll need to clone the tartufo repo and setup poetry!
+
+
+   .. code-block:: console
+
+      git clone git@github.com:godaddy/tartufo.git
+      
+   Development Use Only Poetry Setup: :doc:`CONTRIBUTING` setting-up-a-development-environment
```

Comment:
I think this should work to link directly to the appropriate anchor, and also give nice link text.
```suggestion
   Development Use Only Poetry Setup: :doc:`Setting up a development environment <CONTRIBUTING#setting-up-a-development-environment>`
```

Reply:
I'm liking this section a whole lot more now. This is great!

Looking at it again with fresh eyes has got me wondering: Since it's optional, should it be a part of the numbered bullet points, or would it perhaps serve better (and look better!) as a "note" section? I think it would be interesting to at least try and see what it looks like. It should be very similar to the "Warning" here, but using the light blue colors and say "Note": https://tartufo.readthedocs.io/en/stable/usage.html#python-pre-commit-hook

And theoretically all you have to do to accomplish that is put a `.. note::` line before this, and remove the `#.` so it's no longer a bullet point.

Worth a look, you think?

Reply:
adding `.. note::` seems to conflict with the code-block section just below

Reply:
Looks great here when I render it locally! Aside from the "Contributing" link being still a bit off. Or rather, the text after it looking out of place.
<img width="704" alt="Screen Shot 2020-09-14 at 5 16 46 PM" src="https://user-images.githubusercontent.com/37809/93139180-40324480-f6ae-11ea-94b3-44eb9b224c9f.png">

Reply:
Also, I found a solution to the link problem, here: https://stackoverflow.com/a/19543591/1971587

It's not ideal, but I realized it's because the `CONTRIBUTING` is a `.md` file still. And that's preventing some of the built-in mechanisms from working. That should probably get converted at some point. Regardless, this works for now:

```suggestion
   Development Use Only Poetry Setup: `Setting up a development environment <CONTRIBUTING.html#setting-up-a-development-environment>`_
```

Reply:
ah-ha! thanks for the tip!